src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/client/RestClient.java

```diff
@@ -96,4 +106,33 @@ public Agreement getApplicableAgreement(String tenantId, String personId, LocalD
 
     return getAgreementById(tenantId, agreementId);
   }
+
+  public List<Accrual> patchAccruals(String tenantId, List<Accrual> accruals) {
+    Map<String, String> parameters = Map.of(TENANT_ID_STRING_IDENTIFIER, tenantId);
+
+    List<PatchBody> payloadBody = createPatchBody(accruals);
+    HttpEntity<List<PatchBody>> request = new HttpEntity<>(payloadBody);
+
+    ResponseEntity<ApiResponse<Accrual>> entity = restTemplate.exchange(accrualsNoFilterUrl,
+        HttpMethod.PATCH, request, new ParameterizedTypeReference<>() {
+        }, parameters);
+
+    return Objects.requireNonNull(entity.getBody()).getItems();
+  }
+
+  List<PatchBody> createPatchBody(List<Accrual> accruals) {
+    List<PatchBody> body = new ArrayList<>();
+
+    accruals.forEach(a -> {
+      PatchBody blob = new PatchBody();
+
+      blob.setOp("replace");
+      blob.setPath("/" + a.getId().toString());
+      blob.setValue(a);
+
+      body.add(blob);
+    });
+
+    return body;
```

Comment:
You may have a different preference. but using stream would shave of a few lines of code:

```
    return accruals.stream()
        .map(a -> {
          PatchBody blob = new PatchBody();

          blob.setOp("replace");
          blob.setPath("/" + a.getId().toString());
          blob.setValue(a);

          return blob;
        })
        .toList();
```

Reply:
Updated